doc/website/concepts/qos-policies.md

```diff
@@ -0,0 +1,33 @@
+# QoS policies
+
+Unlike other middlewares, e.g. [DDS used in ROS 2](https://docs.ros.org/en/galactic/Concepts/About-Quality-of-Service-Settings.html),
+iceoryx keeps things simple and offers basic Quality of Service (QoS) settings. The QoS
+settings are called options and can be used to optimize and tailor the communication.
+
+## `PublisherOptions` and `SubscriberOptions`
+
+The three most important settings are:
+
+| Option                              | Explanation                                                                                                                                       |
+|-------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|
+| `PublisherOptions::historyCapacity` | This enables late-joining subscribers to request the n last samples                                                                               |
+| `SubscriberOptions::queueCapacity`  | Size of the subscriber queue where samples are stored before they are passed to the user                                                          |
+| `SubscriberOptions::historyRequest` | The number of samples a late-joining subscriber will request from a publisher, should be equal or smaller than `historyCapacity` otherwise capped |
```

Comment:
I had a brief look to check what happens when `historyRequest` is larger than `queueCapacity` and it seems there is either
- safe overflowing for `QueueFullPolicy::DISCARD_OLDEST_DATA`, i.e. latest gratest
- newer samples are discarded when  `QueueFullPolicy::BLOCK_PUBLISHER`, i.e. lost samples with a gap
Could you please check if I'm right?

For the first case I think it's the right behavior. For the second one I would be surprised. Since RouDi puts the history into the queue, we cannot block RouDi. In order to prevent a gap in the samples, I would suggest to either:
- catch this in the runtime when a subscriber is created
    - either saturate `historyRequest` value to `queueCapacity` and print a warning
    - or abort the creation of the subscriber
- catch it in ChunkDistributor::tryAddQueue and push only the newest samples to the queue to make it full
    - this should also be done as optimization for  `QueueFullPolicy::DISCARD_OLDEST_DATA` since we know that there will be an overflow

@budrus what do you think?

Reply:
If we say request 5 samples (and there are 5 available) and the queue can only hold three I would think we should store the latest 3 (i.e. sample 3 to 5). This is just happens automatically if we are requesting all and discard the first 2. 

Now this only holds in isolation, for multi-publish we may receive interleaved other samples of a different publisher and hence lose some of the three samples. This is not really tied to history but how multi-publish currently works (see above).

This is all consistent with the `QueueFullPolicy::DISCARD_OLDEST_DATA` I would think (as does @elBoberido as far as I understand). 

Now if we let the publisher block if the queue is full, I would expect the publisher to send the first 3 samples and then refuse sending more (to any subscriber). Whether this is desirable is another matter. This again does not consider the effects of multi-publish (which may cause other samples to be in the queue before blocking at 3 samples).

One could argue that such a setting of hsitroyRequest > queueCapacity should be illegal (which would make things easier), but the semantics above are reasonable. Currently those settings are valid. The details of the behavior and *intended semantics* are not entirely clear to me (needs further analysis).

Reply:
@mossmaurice could you please check whether my assumption from the second bullet point of the first list is correct? If yes, there is another limitation for the history when `QueueFullPolicy::BLOCK_PUBLISHER` is used

Reply:
> newer samples are discarded when QueueFullPolicy::BLOCK_PUBLISHER, i.e. lost samples with a gap
Could you please check if I'm right?

@elBoberido This can't happen anymore as third matching criteria was introduced by @MatthiasKillat `SubscriberOptions::historyRequest <= PublisherOptions::historyCapacity`.

Reply:
> newer samples are discarded when QueueFullPolicy::BLOCK_PUBLISHER, i.e. lost samples with a gap
Could you please check if I'm right?

@elBoberido You're right. I adapted `iceoptions` and there is indeed a gap, when starting the subscriber late:

```
publisherOptions.historyCapacity = 10U;
subscriberOptions.queueCapacity = 1U;
-----
iox-cpp-subscriber-with-options got value: 4
iox-cpp-subscriber-with-options got value: 9
iox-cpp-subscriber-with-options got value: 10
iox-cpp-subscriber-with-options got value: 11
iox-cpp-subscriber-with-options got value: 12
iox-cpp-subscriber-with-options got value: 13
iox-cpp-subscriber-with-options got value: 14
iox-cpp-subscriber-with-options got value: 15
```

I documented that as a limitation as the code freeze for `v2.0`. Is that ok from your end?